src/core/main.c

```diff
@@ -273,6 +273,14 @@ _noreturn_ static void crash(int sig) {
                         siginfo_t status;
                         int r;
 
+                        if (siginfo) {
+                                _cleanup_free_ char *cmdline = NULL;
+                                pid_t sender_pid = siginfo->si_pid;
+
+                                (void) get_process_cmdline(sender_pid, SIZE_MAX, 0, &cmdline);
+                                log_emergency("Caught <%s> from PID "PID_FMT" (%s)", signal_to_string(sig), sender_pid, strna(cmdline));
+                        }
```

Comment:
note that the sender_pid can also be zero, for example if we are killed from another process outside of our own PID namespace. here it would be displayed as if we killed outselves, since get_process_cmldine()  reads our own cmdline if pid is passed as zero. Thus we should detect this case, and generate a different message instead.

Hence maybe:

```c
if (siginfo->si_pid == 0)
        log_emergency("Caught <%s> from unknown sender process …
else if (siginfo->si_pid == 1)
        log_emergency("Caught <%s> from our own process …
else
        log_emergency("Caught <%s> from PID " PID_FMT …
```

you get the idea.